spharpy/spherical.py

```diff
@@ -1,94 +1,455 @@
+            self.compute_inverse()
+        return np.dot(self.basis_inv.T, coefficients)
+
+    def inverse_transform_gradient(self, coefficients):
+        if self.basis_inv_gradient_theta is None or self.basis_inv_gradient_phi is None:
+            self.compute_inverse_gradient()
+        return np.dot(self.basis_inv_gradient_theta, coefficients), np.dot(self.basis_inv_gradient_phi, coefficients)
+
+    def interpolate(self, non_uniform_coords):
+        # Implement interpolation algorithm using the basis matrix
+        # TODO: Implement interpolation algorithm using the basis matrix
+        pass
+
+    def filter(self, coefficients, band_type, cutoff_order):
+        # Implement filtering algorithm using the basis matrix
+        # TODO: Implement filtering
+        pass
+
+    def plot_basis_functions(self, order, mode):
+        # Implement plotting function for basis functions
+        # TODO: Implement plotting function for basis functions
+        pass
+
+    def plot_reconstructed_signal(self, coefficients):
+        # Implement plotting function for reconstructed signal
+        # TODO: Implement plotting function for reconstructed signal
+        pass
+
+    @basis.setter
+    def basis(self, value):
+        self._basis = value
+
+    @basis_gradient_theta.setter
+    def basis_gradient_theta(self, value):
+        self._basis_gradient_theta = value
+
+    @basis_gradient_phi.setter
+    def basis_gradient_phi(self, value):
+        self._basis_gradient_phi = value
+
+
+def to_maxN_norm(acn):
+    """
+    Converts N3D normalization to max N normalization
     Parameters
     ----------
-    acn : ndarray, int
-        Linear index
+    acn : int
 
+    Returns
+    -------
+    maxN : float
+        Maximum norm for spherical harmonics of order N
+    """
+    assert acn <= 15
+    maxN = [np.sqrt(1 / 2),
+            np.sqrt(1 / 3),
+            np.sqrt(1 / 3),
+            np.sqrt(1 / 3),
+            2 / np.sqrt(15),
+            2 / np.sqrt(15),
+            np.sqrt(1 / 5),
+            2 / np.sqrt(15),
+            2 / np.sqrt(15),
+            np.sqrt(8 / 35),
+            3 / np.sqrt(35),
+            np.sqrt(45 / 224),
+            np.sqrt(1 / 7),
+            np.sqrt(45 / 224),
+            3 / np.sqrt(35),
+            np.sqrt(8 / 35)]
+    return maxN[acn]
+
+
+def to_sn3d_norm(m, n):
+    """
+    Converts N3D normalization to SN3D normalization
     Parameters
     ----------
-    n : ndarray, int
-        Spherical harmonic order
-    m : ndarray, int
+    m : int
         Spherical harmonic degree
+    n : int
+        Spherical harmonic order (n >= |m|)
 
+    Returns
+    -------
+    sn3d : float
+        SN3D normalization factor
     """
-    acn = np.asarray(acn, dtype=int)
-
-    n = (np.ceil(np.sqrt(acn + 1)) - 1)
-    m = acn - n**2 - n
+    return 1 / np.sqrt(2 * n + 1)
 
-    n = n.astype(int, copy=False)
-    m = m.astype(int, copy=False)
-
-    return n, m
 
-
-def nm2acn(n, m):
+def fuma2nm(fuma):
```

Comment:
probably make it similar to above methods "to_xxx" instead of "2xxx". So renaming to fume_to_nm would be more consistent to me.